Add single forecast GOCART support

WalterKolczynski-NOAA opened the pull request:
**Description**

Adds support for running single coupled free forecasts with GOCART aerosols. Support for continuous aerosol fields from cycle to cycle will be added in a future PR.

To turn on aerosols, there is a new `--aerosols` option for `setup_expt.py`. The option takes no arguments. When it is used, aerosols will be turned on by setting `DO_AERO="YES"` in `config.base`.

GOCART output files are placed in the `chem` directory in COM.

There is a new tarball archived, `chem.tar`, that contains the GOCART output.

Partially addresses #516
Fixes #403 

**Type of change**

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

**How Has This Been Tested?**

- [x] Forecast-only coupled test on Hera
- [x] Forecast-only atm-gocart test on Hera
- [x] Cycled test on Hera
- [x] Forecast-only coupled test on WCOSS-Dell
- [x] Forecast-only atm-gocart test on WCOSS-Dell
- [x] Cycled test on WCOSS-Dell
- [x] Forecast-only coupled test on Orion
- [x] Forecast-only atm-gocart test on Orion
- [x] Cycled test on Orion
  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

WalterKolczynski-NOAA committed:
Add aerosol files to archive job

Refs: #516

WalterKolczynski-NOAA committed:
Enable connectionless UDP for MPI

A new MPI setting to turn on connectionless UDP. This was found to
be necessary for machine stability when aerosols are turned on
(specifically for MAPL). Per "Optimizing HPC Applications with
Intel Cluster Tools: Hunting Petaflops", this setting should be on
when running with thousands of processes.

When we break the modulefiles apart by job, we might want to have
this change only apply to the forecast job.

Refs: #516

WalterKolczynski-NOAA committed:
Remove ADJUST_ALLREDUCE setting on Orion

The setting had already been removed from Hera and is not present
for other machines.

Refs: #516

WalterKolczynski-NOAA committed:
Link aerosol output files

Expected aerosol output files are now linked to their final position
on COM in the same manner as other components.

Refs: #516

WalterKolczynski-NOAA committed:
Add --aerosols option to setup expt

To control whether coupled aerosols are used, a new option is added
to the experiment setup scripts: --aerosols. It is an optional switch
and takes no positional arguments. If --aerosols is used, DO_AERO will
be set to "YES"; otherwise it will be "NO".

The ATMAERO app is removed as an option in config.base as redundant.
The setting had never been added to the setup expt scripts.

Refs: #516

WalterKolczynski-NOAA committed:
Increase stack size for coupled

The domain stack size is doubled in the default coupled settings.
The larger value is needed when GOCART is used.

Refs: #516

WalterKolczynski-NOAA committed:
Update diag table for coupled aerosols

Now that aerosols are permitted to run with coupled, a new diag table
is needed for that configuration. Instead of adding nested ifs to
handle the different combinations, the coupled diag table is moved
out into its own block, then if aerosols are on, '_aer' is appended to
the file name, whichever diag table is used.

In the future, I hope to break diag_table up into pieces that can be
assembled based on component (see Issue #567).

Refs: #516

WalterKolczynski-NOAA committed:
Add new nems configure for coupled aerosols

A new nems.configure template is added for coupled aerosols. The
allowed combinations of components is updated in cplvalidate to
allow the new configuration.

Refs: #516

WalterKolczynski-NOAA committed:
Update nems.configure for atm-only aerosols

The nems.configure file for atm-only needed to be updated for the
current version of UFS.

Refs: #516

WalterKolczynski-NOAA committed:
Add aerosol output files to archive job

A new list of files (chem.txt) is created to archive. The files is
already being looked for by the arch.sh.

Refs: #516

WalterKolczynski-NOAA committed:
Rearrange when aerosol PETs are set

Moved the aerosol PETs definition up so it is next to where the value
the atm PETs are defined (since it uses the same PETs) and clarified
the comments. There should be no functional change.

Refs: #516

WalterKolczynski-NOAA committed:
Build UFS with GOCART

The UFS build script is updated to always build with GOCART available.

Refs: #516

rmontuoro committed:
Set nems.configure template for S2SW in EMC's version of config.base.

rmontuoro committed:
Consolidate logical coupling flags for aerosols (cplgocart, cplchem) into cplchm.

rmontuoro committed:
Remove unnecessary statement.

rmontuoro committed:
Properly set nems.configure template for S2SW in EMC installation.

rmontuoro committed:
Fix DO_AERO test.

rmontuoro committed:
Use _aero as suffix for nems.configure template file including aerosols.

rmontuoro committed:
Implement prognostic aerosols (UFS-Aerosols).

Default aerosol settings are included in parm/config/config.aero
and GOCART resource files in parm/chem. These files may be edited
to customize the model configuration before setting up the workflow.

The GBBEPx and QFED options are provided to select input fire emissions
via the AERO_EMIS_FIRE configuration variable in config.aero.
If no option is provided, the workflow will default to NONE (no fires).

rmontuoro committed:
Move field_table and diag_table GOCART fragments to parm/parm_fv3diag.

rmontuoro committed:
Remove previous set of aerosol diagnostic and field tables.

rmontuoro committed:
Set AERO_CONFIG_DIR to default value in config.aero.

rmontuoro committed:
Replace short-circuit construct with if block.

rmontuoro committed:
Consolidate aerosol variable names.

rmontuoro committed:
Introduce fix files location for UFS-Aerosols (GOCART).

WalterKolczynski-NOAA committed:
Fix link to GOCART fix directory

Updated the link created to the GOCART fix directory in $DATA. It was
previously being linked as fix_gocart, but ExtData is used in all the
GOCART rc files.

Refs: #516

WalterKolczynski-NOAA committed:
Merge pull request #17 from rmontuoro/feature/p8-aerosols

Add prognostic aerosols

WalterKolczynski-NOAA committed:
Merge branch 'develop' into feature/aerosols

WalterKolczynski-NOAA committed:
Link gocart fix files

The gocart fix directory is added to the list of directories linked by
the link script.

Refs #516

WalterKolczynski-NOAA commented:
Ran a full 16-day coupled test on Hera, which completed after increasing forecast walltime:

Forecast run directory: `/scratch2/NCEPDEV/ensemble/noscrub/Walter.Kolczynski/ptmp/RUNDIRS/aerosols/2013040100/gfs/fcst.181907`
COM: `/scratch2/NCEPDEV/ensemble/noscrub/Walter.Kolczynski/global-workflow/aerosols/gfs.20130401/00`
Archive: `/NCEPDEV/emc-global/1year/Walter.Kolczynski/HERA/scratch/aerosols/2013040100`

Will run more tests, but this is probably the one to verify for the technical aspects.

JessicaMeixner-NOAA commented on parm/config/config.resources:
```diff
@@ -201,6 +201,12 @@ elif [ $step = "fcst" ]; then
 
+        if [[ $DO_AERO == "YES" ]]; then
+            # Aerosol model only uses the atm model PETS
+            export CHMPETS=$ATMPETS
```
Is this ATMPETS or MEDPETS?  In the regression tests we've been setting up we've not used the write grid component ATM PETs so therefore have been using just the MEDPETS... What's the best thing here @rmontuoro ?

WalterKolczynski-NOAA replied:
This is before the write groups get added, so it is the same as MEDPETS

JessicaMeixner-NOAA commented on parm/config/config.base.emc.dyn:
```diff
@@ -179,21 +179,20 @@ case "${APP}" in
 esac
 
+if [[ $DO_AERO = "YES" ]]; then
+    export confignamevarfornems="${confignamevarfornems}_aero"
```
Is this the naming convention we want to use?   I'm working on updates to ufs-weather-model to add this to the regression tests and this is not the naming convention I used: https://github.com/JessicaMeixner-NOAA/ufs-weather-model/blob/feature/p8b_aero/tests/parm/nems.configure.cpld_aero_wave.IN I'm fine with either (I went alphabetical aero_wave instead of wave_aero), but we should be consistent between the two to make things simpler.  So we should either change that here or change that in my branch.

WalterKolczynski-NOAA replied:
Because of the way this is currently set up (waves are turned on as part of the APP setting, but aerosols can be tacked on to anything), it would be a lot easier to keep it as it is in this PR.

JessicaMeixner-NOAA replied:
Okay, but this would be a 3 line change to change this here and then renaming the nems.congifure file versus having to change this in 20+ different test descriptions in rt.sh + renaming all the files in my branch of the workflow for p8 plus the ufs- branch.   So if logically this is a better reason, okay, but it's non-trivial to change this on my end so I want to make sure this is really what we want to do before I spend the time changing it on my end.

rmontuoro replied:
We can certainly modify the template names and add a test for waves following the aerosol one:
```
if [[ $DO_WAVE = "YES" ]]; then
  export confignamevarfornems="${confignamevarfornems}_wave
fi
```

Ideally, this could be streamlined as follows:
```
for comp in ${COMPONENTS} ; do
  export confignamevarfornems="${confignamevarfornems}_${comp}
done
```
where `COMPONENTS` is an ordered list of lowercase component names. This construct can be extended to other portions of the global workflow, also using `case ` if needed:
```
for comp in ${COMPONENTS} ; do
  case ${comp} in
    aero)
      ... process aerosol options ...
      ;;
    wave)
      ... process wave options ...
      ;;
    ...
  esac
done
```

JessicaMeixner-NOAA replied:
@WalterKolczynski-NOAA any thoughts on this?  I need to know so if I have to change my ufs-waether-model work I can start that ASAP as there will be a lot to change and test.

WalterKolczynski-NOAA replied:
Let's go ahead with your convention. I'll change the PR.

JessicaMeixner-NOAA replied:
Thanks Walter - let me know if you need anything from me, I have some of these updates here as well, if that's helpful: https://github.com/jessicameixner-noaa/global-workflow/tree/feature/p8b. Let me know if there's anything I can do to help with this change.

JessicaMeixner-NOAA commented on parm/config/config.defaults.s2sw:
```diff
@@ -47,3 +47,6 @@ IOSRV=1
 export ARCH_GAUSSIAN_FHINC=${FHOUT_GFS}
+
+# for forecast namelist
+export domains_stack_size="6000000"
```
Is this the domain stack size for either with or without aero?  Because when we have aerosols this needs to be much bigger, but if you do not, I think it's fine to use the smaller amount?

WalterKolczynski-NOAA replied:
Only gocart needs it, but I don't think it hurts to use the same without it.

JessicaMeixner-NOAA replied:
Actually this needs to be changed for Thompson MP even without GOCART and then if you add GOCART it needs to be even larger, so I think having it dependent on GOCART is a good idea.

JessicaMeixner-NOAA replied:
But perhaps I misunderstand it's total impact when we add Thompson MP

WalterKolczynski-NOAA replied:
Increased to 16000000 for the time being.

yangfanglin commented:
I'd suggest to use domains_stack_size = 8000000  for all cases.  I have been using this setting for running C768L127 with updated phsyics; otherwise, the model will blow up.   Running the coupled model at C768L127 with aerosol may require an even larger number.  But that needs to be tested and decided in the future.  
It appears using a larger number makes no harm to forecast speed.  It might be architecture depend.

JessicaMeixner-NOAA commented:
> I'd suggest to use domains_stack_size = 8000000 for all cases. I have been using this setting for running C768L127 with updated phsyics; otherwise, the model will blow up. Running the coupled model at C768L127 with aerosol may require an even larger number. But that needs to be tested and decided in the future.
> It appears using a larger number makes no harm to forecast speed. It might be architecture depend.

That's too small for ThompsonMP+aerosols

JessicaMeixner-NOAA commented:
We need domains_stack_size=16000000 (or at least > 8000000) for C192 and C384 for ThompsonMP+Aerosols.  I came up with that number b/c before +aerosol doubled it.  It might could be smaller, but I didn't test that.

SMoorthi-emc commented:
I have been using "10240000" as default and have been able to run C1152L127
without any problem.
Moorthi

On Fri, Feb 18, 2022 at 2:49 PM Jessica Meixner ***@***.***>
wrote:

> We need domains_stack_size=16000000 (or at least > 8000000) for C192 and
> C384 for ThompsonMP+Aerosols. I came up with that number b/c before
> +aerosol doubled it. It might could be smaller, but I didn't test that.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/global-workflow/pull/659#issuecomment-1045088046>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYR4TPPGXJWKCMI2AELU32PDRANCNFSM5OXR6HAA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

JessicaMeixner-NOAA commented:
> I have been using "10240000" as default and have been able to run C1152L127 without any problem. 

@SMoorthi-emc is that with GOCART?

SMoorthi-emc commented:
No, I have no intention of trying GOCART (I don't think you can run it at
C1152 127 anyway).

On Fri, Feb 18, 2022 at 3:12 PM Jessica Meixner ***@***.***>
wrote:

> I have been using "10240000" as default and have been able to run
> C1152L127 without any problem.
>
> @SMoorthi-emc <https://github.com/SMoorthi-emc> is that with GOCART?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/global-workflow/pull/659#issuecomment-1045120861>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYRU2VQFMWN7Z4PSFNTU32R4FANCNFSM5OXR6HAA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: ***@***.***
Phone:  (301) 683-3718   Fax: (301) 683-3718

rmontuoro commented on sorc/build_ufs.sh:
```diff
@@ -5,17 +5,11 @@ set -eux
-while getopts "ac" option; do
+while getopts "c" option; do
   case "${option}" in
-    a)
```
Note that the atmosphere-aerosol coupling configuration should be supported for aerosol tuning and evaluation runs. While we can still build the UFS weather model including all coupled components, we may not be able to use the same CCPP physics suites and atmosphere settings for atmosphere-aerosol and fully coupled configurations.

Note also that cycled aerosol runs are performed only with atmosphere-aerosol configurations.

WalterKolczynski-NOAA replied:
This switch is going to go away entirely in a separate PR. I tested and I can run coupled & atm-only from the same executable. To start, I am going to compile with all the CCPP suites from both, then we can evaluate which are needed.

rmontuoro reviewed (approved):
Looks good. Just a few comments.

WalterKolczynski-NOAA committed:
Rename atm-aero template and add error checking

The nems.configure template for atm-aerosols was not update from aer to aero,
causing the forecast to fail for that configuration.

Also added some error checking to catch the error earlier if the file is missing
and report the problem without having to dig into PET logs (which often aren't
even produced).

Refs: #516

WalterKolczynski-NOAA committed:
Increase stack size for all forecast

The stack size is increased for all forecasts to accomodate both
aerosols and looking forward to Thompsan microphysics. This means
the special override for coupled (which was also for aerosols) could
be removed.

Refs: #516

WalterKolczynski-NOAA committed:
Merge branch 'develop' of github.com:NOAA-EMC/global-workflow into feature/aerosols

WalterKolczynski-NOAA committed:
Turn off cplflx for aerosols

Turns off the cplflx flag, which is not needed for just atm-aerosols.

Refs: #516

WalterKolczynski-NOAA committed:
Only substitute DO_AERO for forecast-only experiments

The setup experiment script was attempting to replace the DO_AERO variable
in config.base using inputs.aerosols, but that option is only defined for
forecast-only, causing a failure due to a key exception for cycled experi-
ments. Now the substitution is only attempted for forecast-only experiments.

Refs: #516

WalterKolczynski-NOAA committed:
Reorder components in NEMS template

Reorders the components in the nems.configure files to alphabetical
order to match conventions being applied to UFS.

Refs: #659

WalterKolczynski-NOAA committed:
Move aerosol emission input out of fix

The aerosol emission files aren't really fix files. At some point down
the line, they will come from an outside source like DCOM for real-time
forecasts, so placing them in fix isn't appropriate. Instead the inputs
currently being used for prototypes are being staged elsewhere on each
HPC. The directory set in config.aero is updated to point the approriate
location based on machine instead of a fix directory.

Ref: #516

WalterKolczynski-NOAA committed:
Merge branch 'develop' into feature/aerosols

* develop:
  Cleanup of utils directory (#660)
  Fix bug with partition_batch on WCOSS (#668)

rmontuoro committed:
Add time template for FENGSHA's time-varying sediment supply map (SSM)
input to aerosol resource file template (ExtData).

CoryMartin-NOAA commented:
General comment, why is everything `chem` when this is the `aero` component? GOCART is not reactive chemistry like CMAQ, and we should be careful to distinguish between the two. As RRFS will have `chem` or `aq` which will be a different modeling component.

WalterKolczynski-NOAA commented:
> General comment, why is everything `chem` when this is the `aero` component? GOCART is not reactive chemistry like CMAQ, and we should be careful to distinguish between the two. As RRFS will have `chem` or `aq` which will be a different modeling component.

A few reasons, though none of them are particularly good.
- We (global-workflow) don't really have control over what they call it within UFS, and they use chm (the setting to turn it on is `cplchm`, the nems.configure template use `chm_model`, etc.)
- Probably relatedly, that is how the AQ team often refer to it
- Also what was used when it was added to coupled-crow (it was more consistently "chem" there)
- GEFS-aerosol also outputs the AQ data to a `chem` directory

That said, I've tried to use `aero` for the most part. The exceptions being the name of the parm directory and the aforementioned output directory in COM, which I should be able to change easily. Now is probably the best time before things become entrenched.

I propose changing the parm directory to `gocart` and the COM directory to either `aq` or `aero`. Thoughts? @JessicaMeixner-NOAA @rmontuoro @KateFriedman-NOAA @jeffmcqueen

JessicaMeixner-NOAA commented:
Will there be an expectation that this is the same COM dir name as what is being done in GEFS already from NCO and others?

jeffmcqueen commented:
We are planning to add gas phase (non aerosol) chemistry.  a proposal to
add simple ozone chemistry (less than 13 species)...i agree with Raffaele's
point that aq is a good choice.

Jeffery T. McQueen
Air Quality and Dispersion Modeling Team Leader
National Center for Weather and Climate Prediction
Room 2095
5830 University Research Court
College Park, MD 20741
301-683-3736
Fax: 301-683-3703


On Wed, Mar 2, 2022 at 2:50 AM Walter Kolczynski - NOAA <
***@***.***> wrote:

> General comment, why is everything chem when this is the aero component?
> GOCART is not reactive chemistry like CMAQ, and we should be careful to
> distinguish between the two. As RRFS will have chem or aq which will be a
> different modeling component.
>
> A few reasons, though none of them are particularly good.
>
>    - We (global-workflow) don't really have control over what they call
>    it within UFS, and they use chm (the setting to turn it on is cplchm,
>    the nems.configure template use chm_model, etc.)
>    - Probably relatedly, that is how the AQ team often refer to it
>    - Also what was used when it was added to coupled-crow (it was more
>    consistently "chem" there)
>    - GEFS-aerosol also outputs the AQ data to a chem directory
>
> That said, I've tried to use aero for the most part. The exceptions being
> the name of the parm directory and the aforementioned output directory in
> COM, which I should be able to change easily. Now is probably the best time
> before things become entrenched.
>
> I propose changing the parm directory to gocart and the COM directory to
> either aq or aero. Thoughts? @JessicaMeixner-NOAA
> <https://github.com/JessicaMeixner-NOAA> @rmontuoro
> <https://github.com/rmontuoro> @KateFriedman-NOAA
> <https://github.com/KateFriedman-NOAA> @jeffmcqueen
> <https://github.com/jeffmcqueen>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/global-workflow/pull/659#issuecomment-1056489445>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALPIHKW5V7TTWN5WXSVIIQLU54MSVANCNFSM5OXR6HAA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

bbakernoaa commented:
@WalterKolczynski-NOAA I think that the issue with changing the directory to `aero` from `chem` is kinda semantics really but if you start going down that path what is going to happen when more sophisticated chemistry is added such as what @jeffmcqueen mentioned or if we run UFS-CMAQ or some other chemical model.  I think that it is easiest and more clear that we keep it as `chem` as it allows for expansion rather than being overly expressive. I really just don't see the need to buck what was already there.

CoryMartin-NOAA commented:
My original comment was more of the fact that `parm` is a mix of conventions (`wave` or `mom6`) and that `chem` is too vague, so @WalterKolczynski-NOAA 's suggestion of `parm/gocart` is best here IMO. For the COMDIR, `chem` seems fine from an output directory perspective as it is unlikely that there would be both GOCART and CMAQ (or what Jeff proposed) in the same run, right?

bbakernoaa commented:
@CoryMartin-NOAA That isn't necessarily the case.  For instance, if down the road we decide to turn other components on in the NASA component which includes things like GEOS-CHEM or other components then it would still write out to the chem directory.  I just don't see the need to be more specific with `parm/gocart` or something else when we don't need to. This also limits the number of hard coded sections within the workflow allowing for easier expansion.

CoryMartin-NOAA commented:
@bbakernoaa but aren't most/all of these files in `parm/chem` GOCART specific? If not, then ok, but if yes, I think it is important to organize them (either by name or subdirectory) in a way that identifies them as such. But I'll defer to the actual modelers on decisions here, just wanted to ask questions to make sure things are 'correct' before they are set in stone.

bbakernoaa commented:
@CoryMartin-NOAA Currently they are in the global aerosol model only GOCART.  But if we label it GOCART and we add another component from NASA such as `ACHEM` then it isn't just `GOCART`.  I think that this isn't necessary and is really just a distraction from getting this through.  Let's not add more onto the pile and get something that works so that the component can continue and progress.

CoryMartin-NOAA commented:
@bbakernoaa ok, I'm just worried about having a mess of a `parm/chem` directory in the future, but as you said, this needs merged in ASAP (and then cycling...) so I guess that bridge can be crossed when we get to it.

WalterKolczynski-NOAA committed:
Use correct g2tmpl variable

The nceppost_cpl job was using the wrong g2tmpl variable in the definition
of the grib2 table.

WalterKolczynski-NOAA committed:
Only use serial netcdf for EnKF on Orion

Although it seemed to work previously, parallel netcdf is now causing
the same issues with the ensemble output that has been seen on Hera
and WCOSS2.

WalterKolczynski-NOAA committed:
Merge branch 'develop' into feature/aerosols

* develop:
  Specify memory for init job (#669)

WalterKolczynski-NOAA committed:
Merge pull request #18 from rmontuoro/feature/p8-aerosols

Add time template for FENGSHA's time-varying sediment supply map (SSM)

KateFriedman-NOAA commented on env/ORION.env:
```diff
@@ -17,7 +17,6 @@ export npe_node_max=40
 export launcher="srun -l --export=ALL"
 
 # Configure MPI environment
-export I_MPI_ADJUST_ALLREDUCE=5
```
Why is this setting removed? I looked back at my emails and this was added on Theia during the move from MOAB to SLURM and had to do with reproducibility? Does the model reproduce on Orion without this setting? If so, I'm ok having it removed.

WalterKolczynski-NOAA replied:
@rmontuoro ?

WalterKolczynski-NOAA replied:
Confirmed results of the cycled test are still bitwise identical across two runs.

WalterKolczynski-NOAA committed:
Only turn on DAPL user datagrams for forecast job

Enabling DAPL user datagrams was causing problems in the analysis
on WCOSS for as yet unknown reasons, so the setting was moved out
of the modulefile and into the forecast section of the environment
file. For consistency, this was done for all supported machines
even though the issue only arose on WCOSS.

Refs: #516

JessicaMeixner-NOAA reviewed (approved):
As long as the tests are passing the code updates look good to me.  Thanks for changing the naming convention! I really appreciate that.

aerorahul commented on sorc/build_ufs.sh:
```diff
@@ -50,5 +44,5 @@ if [ -d build ]; then
 fi
 mkdir -p build && cd build
-cmake -DAPP=${APP} -DCCPP_SUITES=${CCPP_SUITES} ..
+cmake -DAPP=${APP} -DCCPP_SUITES=${CCPP_SUITES} -DUFS_GOCART="ON" ..
```
`-DUFS_GOCART` is not an exposed option of the UFS.
The correct way to enable aerosols is using the `-DAPP=ATMAERO`.

aerorahul commented on parm/config/config.base.emc.dyn:
```diff
@@ -149,28 +149,35 @@ export DO_COUPLED="NO"
 export DO_OCN="NO"
 export DO_ICE="NO"
-export DO_AERO="NO"
+export DO_AERO=@DO_AERO@
```
`DO_AERO` should be `NO` by default and be toggled as is done for the other components based on `APP`.

aerorahul reviewed (commented):
The aerosol component in the model is turned on via a model internal flag.
The `aerosols` in the workflow is handled in a different manner than other components.
These need to be reverted ASAP.
